espnet/nets/pytorch_backend/rnn/decoders_transducer.py

```diff
@@ -69,23 +73,35 @@ def __init__(self, eprojs, odim, dtype, dlayers, dunits, blank,
         self.ignore_id = -1
         self.blank = blank
 
-    def zero_state(self, h_pad):
-        return h_pad.new_zeros(h_pad.size(0), self.dunits)
+    def zero_state(self, ey):
+        z_list = [ey.new_zeros(1, ey.size(0), self.dunits)]
+        c_list = [ey.new_zeros(1, ey.size(0), self.dunits)]
+
+        for _ in six.moves.range(1, self.dlayers):
+            z_list.append(ey.new_zeros(1, ey.size(0), self.dunits))
+            c_list.append(ey.new_zeros(1, ey.size(0), self.dunits))
 
-    def rnn_forward(self, ey, z_list, c_list, z_prev, c_prev):
+        return (z_list, c_list)
+
+    def rnn_forward(self, ey, dstate):
```

Comment:
I want docstrings for its args and return values (e.g., type, comment, and shape) because it is getting a little bit complex.

Reply:
Hum, maybe the problem is the way I wrote the method? Because it should be almost the same as before.
Does the following docstring sound clear to you : 

```
    def rnn_forward(self, ey, dstate):
        """RNN Forward                                                                                                                                               
                                                                                                                                                                     
        Args:                                                                                                                                                        
            ey (torch.Tensor): batch of input features (B, Lmax, Emb_dim)                                                                                            
            dstate (tuple): tuple with hidden and cell state (NLayers, B, Hdec)                                                                                      
        Returns:                                                                                                                                                     
            output (torch.Tensor): batch of output features (B, Lmax, Hdec)                                                                                          
            dstate (tuple): tuple with hidden and cell state (NLayers, B, Hdec)                                                                                      
        """
```

Reply:
LGTM. Because this method can be changed in the future (e.g., multi-layer cudnn states), it is nice to clarify what this method does at this time.